### PR TITLE
Change how status type is encoded in the protocol

### PR DESCRIPTION
**Pull request by scheibo**
Not to be merged until the client can handle the new protocol. We might need to also give bot developers a heads up as well? Discussion is all on #5590

### CODE REVIEW

File: server/users.ts [hunk 1]
```diff
@@ -598,13 +598,11 @@ class User extends Chat.MessageContext {
...
 		const identity = this.getIdentity(roomid);
-		const status = this.statusType === 'online' ? '' : '@!';
+		const status = this.getStatus();
 		return `${identity}${status}`;
 	}
 	getStatus() {
-		const statusMessage = this.statusType === 'busy' ? '!(Busy) ' : this.statusType === 'idle' ? '!(Idle) ' : '';
-		const status = statusMessage + (this.userMessage || '');
-		return status;
+		return this.statusType === 'idle' ? '@!' : this.statusType === 'busy' ? '@' : /* online */ '';
```
Comment (TheJetOU) [inline server/users.ts]: Whether Idle should be `@` or `@?` is up on the fence still, I think?
```suggestion
		return this.statusType === 'busy' ? '@!' : this.statusType === 'idle' ? '@' : /* online */ '';
```

Comment (scheibo) [reply server/users.ts]: No, Zarel is pretty vehement about Busy as `@` and Idle as `@!`, so the code as written should correctly reflect his preference.

Comment (TheJetOU) [reply server/users.ts]: Oh, I reverted it in my brain and thought he wanted `@!` for busy and `@` idle, my bad.

Comment (scheibo) [reply server/users.ts]: Yeah, *we* want `@!` for Busy which is why its so confusing, but to Zarel, Busy is ⊥, and I'm tired to arguing.

**Comment by TheJetOU** [comment]
Hm, looks like you'll also need to change:

-  `/cmd userdetails` https://github.com/Zarel/Pokemon-Showdown/blob/master/server/chat-commands.js#L4485-L4486 (it's already broken anyway, because `targetUser.away` is not a valid property in `User`...) to use `targetUser.userMessage` in the `status` key because `getStatus` doesn't return the status anymore.

- `PROTOCOL.md` to reflect the newest changes.

**Pull request closed without merging**